libfwupdplugin/fu-drm-device.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuDrmDevice"
+
+#include "config.h"
+
+#include "fu-bytes.h"
+#include "fu-drm-device.h"
+#include "fu-string.h"
+
+/**
+ * FuDrmDevice
+ *
+ * A DRM device.
+ *
+ * See also: #FuUdevDevice
+ */
+
+G_DEFINE_TYPE(FuDrmDevice, fu_drm_device, FU_TYPE_UDEV_DEVICE)
+
+static void
+fu_drm_device_to_string(FuDevice *device, guint idt, GString *str)
+{
+	FuDrmDevice *self = FU_DRM_DEVICE(device);
+	fu_string_append(str,
+			 idt,
+			 "Status",
+			 fu_display_state_to_string(fu_drm_device_get_state(self)));
+}
```

Comment:
I think you should also capture `enabled` sysfs file

Reply:
And `connector_id`, could be useful for building quirks for the DRM device.

Reply:
both as new properties?

Reply:
Yeah, I think both are potentially useful to have in the toolbelt.

Reply:
is connector ID always an integer?

Reply:
and bonus question -- does the connector_id ever change? i.e. should we save (and parse it to an integer) in _probe() perhaps?

Reply:
It doesn't look stable to me, it's internally `struct ida`.  So maybe it's worth using for a logical ID at best (/dev/dri/cardX for physical ID).

Reply:
Okay, lets grab as required -- it's not querying the hardware and looks inexpensive so I don't think we need to put it in _probe().